compiler/rustc_query_system/src/dep_graph/graph.rs

```diff
@@ -354,7 +355,7 @@ impl<K: DepKind> DepGraphData<K> {
             None => TaskDepsRef::Ignore,
         };
 
-        let result = K::with_deps(task_deps_ref, || task(cx, arg));
+        let result = K::with_deps(task_deps_ref, || __rust_begin_short_backtrace(|| task(cx, arg)));
```

Comment:
I think there should only be 2 places for these functions:
- one in the `QueryEngine::$name` functions, which is the entry point for the computation;
- one in `QueryConfig::compute` functions, which is where we give back to use code.

Both are in `rustc_query_impl::plumbing`.

Reply:
It's counterintuitive, because of the code logic here: 
https://github.com/rust-lang/rust/blob/f37f8549940386a9d066ba199983affff47afbb4/library/std/src/sys_common/backtrace.rs#L72-L93

We test function sym from bottom to up, so to reduce as much part of backtrace from query system, the `__rust_begin_short_backtrace` should be placed at the innermost function of query system.

I found the positions for these two functions in this way, follow the full log from up to bottom, find the first function we need to mark as `begin`, and find the second function we need to mark as `end`.

![image](https://user-images.githubusercontent.com/230646/224552091-d7ee2a0c-c04f-4d58-80c5-63e239e904b6.png)

Reply:
I mean: the innermost is `QueryConfig::compute`, and the outermost is `QueryEngine::$name`. That's the only 2 places where we need those functions.

Reply:
ah, this makes the changes smaller and more consistent, thanks!